mariamihaelafainarea/assignment4/Assignment4

```diff
@@ -0,0 +1,78 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment4;
+
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ *
+ * @author maria
+ */
+public class Assignment4 {
+
+    int n,m;
+    boolean[][] matrix;
+    DijointSet d;
+    //false - water
+    //true - land
+    //Please return number of islands ! 
+
+    public Assignment4(int n, int m, boolean[][] matrix) {
+        this.n = n;
+        this.m = m;
+        this.matrix = matrix;
+    }
+    //First, assign for each cell a different set
+    public void putIntoDifferentSets() {
```

Comment:
I think you can simplify your algorithm a lot. Try for example combining first two steps: you don't need to first mark all land cells as separate sets and then find which ones belong to the same island, you can do both of those steps on the first traversal of the grid.

Reply:
I do not understand this comment. I cannot apply flood fill until i do not mark all the cells as beeing in different trees.